src/Common/StringSearcher.h

```diff
@@ -587,9 +589,14 @@ class StringSearcher<true, ASCII> : public StringSearcherBase
 
         first = *needle;
 
-#ifdef __SSE4_1__
-        pattern = _mm_set1_epi8(first);
 
+#ifdef __SSE4_1__
+        first_pattern = _mm_set1_epi8(first);
+        if (needle + 1 < needle_end)
+        {
+            second  = *(needle + 1);
+            second_pattern = _mm_set1_epi8(second);
+        }
```

Comment:
Unrelated, but in l. 602, why not simply `for (int i=0; i<n; ++i)`?

Reply:
Done. Simplify the original code with a simple loop.